Orleans.Serialization.NodaTime/CalendarSystemCodec.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Buffers;
+using System.Text;
+using NodaTime;
+using Orleans.Serialization.Buffers;
+using Orleans.Serialization.Codecs;
+using Orleans.Serialization.WireProtocol;
+
+namespace Orleans.Serialization.NodaTime;
+
+[RegisterSerializer]
+public class CalendarSystemCodec : IFieldCodec<CalendarSystem?>
+{
+    public void WriteField<TBufferWriter>(
+        ref Writer<TBufferWriter> writer,
+        uint fieldIdDelta,
+        Type expectedType,
+        CalendarSystem? value)
+        where TBufferWriter : IBufferWriter<byte>
+    {
+        // If the value is null, write it as the null reference.
+        if (value is null)
```

Comment:
Use this instead:

``` csharp
if (ReferenceCodec.TryWriteReferenceField(ref writer, fieldIdDelta, expectedType, value))
{
  return;
}
```

You should delete the `MarkValueField` line below, too, since this is a reference typed value.